generate.coffee

```diff
@@ -118,11 +147,30 @@ generateOrders = (total, customers, addresses, products) ->
       coupon_code: couponCode
       base_tax_amount: (0.08 * grandTotal).toFixed(2)
       base_shipping_amount: shippingAmount
+      base_discount_amount: discountAmount
+      utm_source: utmSource
+      utm_medium: utmMedium
+      utm_campaign: utmCampaign
       created_at: createdAt
       updated_at: createdAt
     orders.push(order)
+    orderCounts[customer.entity_id] = if orderCounts[customer.entity_id] then orderCounts[customer.entity_id] + 1 else 1
   return orders
 
+getCustomerToBuy = (customers, orderCounts) ->
```

Comment:
Were you able to verify that this algorithm creates the proper incrementing event probability?